src/workspace/mod.rs

```diff
@@ -131,6 +133,8 @@ pub enum WorkspaceDiagnosticKind {
   ImportMapReferencingImportMap,
   #[error("Patching npm packages ({package_json_url}) is not implemented.")]
   NpmPatchNotImplemented { package_json_url: Url },
+  #[error("\"nodeModulesDir\" field is deprecated in deno 2.0. Use `\"nodeModules\": \"{0}\"` instead")]
+  DeprecatedNodeModulesDirOption(&'static str),
```

Comment:
Nitpick: Maybe use `NodeModulesMode` here instead then add a display implementation for `NodeModulesMode` to display it properly in the error message? It might help out later and keeps the strings for that closer to the `NodeModulesMode` declaration.